lib/wallet/src/Cardano/Wallet/Submissions/Properties/Primitives.hs

```diff
@@ -0,0 +1,141 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# OPTIONS_GHC -Wno-incomplete-patterns #-}
+
+module Cardano.Wallet.Submissions.Properties.Primitives
+    ( properties
+    , txIds
+    )
+where
+
+import Prelude
+
+import Cardano.Wallet.Submissions.Primitives
+    ( Primitive (..) )
+import Cardano.Wallet.Submissions.Properties.Common
+    ( Step (Step), forAllIn, that, verify, ( # ) )
+
+import Cardano.Wallet.Submissions.Submissions
+    ( Submissions, finality, tip, transactions )
+import Cardano.Wallet.Submissions.TxStatus
+    ( HasTxId (..)
+    , TxStatus (Expired, InLedger, InSubmission, Unknown)
+    , status
+    )
+import Data.Set
+    ( Set, singleton )
+import Test.QuickCheck
+    ( Property, counterexample, (.&.), (===) )
+
+import qualified Data.Map.Strict as Map
+
+txIds :: Submissions slot tx -> Set (TxId tx)
+txIds = Map.keysSet . transactions
+
+-- | Translations of primitive properties from specifications.
+properties
+    :: (Ord (TxId tx), Eq tx, HasTxId tx
+        , Ord slot, Show (TxId tx), Show slot, Show tx)
+    => Step Primitive slot tx -> Property
+properties (Step xs xs' (AddSubmission expiring x)) = do
+    let world = txIds xs <> txIds xs' <> singleton (txId x)
+    counterexample "on add-submission" $ verify $ do
+        that "tip and finality are not changed"
+            $ tip xs === tip xs'
+            .&. finality xs === finality xs'
+        that "finality should never change"
+            $ finality xs === finality xs'
```

Comment:
```suggestion
    counterexample "on add-submission" $ conjoin
        [ that "tip and finality are not changed"
            $ tip xs === tip xs'
            .&. finality xs === finality xs'
        , that "finality should never change"
            $ finality xs === finality xs'
```

🤔 I'd prefer an explicit list, but I'd be happy to keep the `that`.

Likewise for the other occurrences.